examples/profile.html

```diff
@@ -0,0 +1,307 @@
+
+          let clientId = pcOAuthClientIds[pcEnvironment];
+          if (!clientId) {
+              setErrorState(pcEnvironment + ': Unknown/Unsupported PureCloud Environment');
+              return;
+          }
+
+          let platformClient = window.require('platformClient');
+          let client = platformClient.ApiClient.instance;
+          client.setEnvironment(pcEnvironment);
+
+          let clientApp = null;
+          try {
+            clientApp = new window.purecloud.apps.ClientApp({
+                pcEnvironment: pcEnvironment
+            });
+          } catch (e) {
+              setErrorState(pcEnvironment + ': Unknown/Unsupported PureCloud Embed Context');
+              return;
+          }
+
+          let redirectUrl = window.location.origin;
+          if (!redirectUrl) {
+              redirectUrl = window.location.protocol + window.location.host;
+          }
+          redirectUrl += window.location.pathname;
+
+          // Authenticate with PureCloud
+          let authenticated = false;
+          let authenticatingEl = document.querySelector('.authenticating');
+          setHidden(authenticatingEl, false);
+          client.loginImplicitGrant(clientId, redirectUrl, {state: ('pcEnvironment=' + pcEnvironment)})
+            .then(function() {
+                authenticated = true;
+
+                return new platformClient.UsersApi().getUsersMe();
+            })
+            .then(function(profileData) {
+                setHidden(authenticatingEl, true);
+
+                let profileEl = document.querySelector('.user-profile');
+                renderUserProfile(profileEl, profileData);
+                setHidden(profileEl, false);
+            })
+            .catch(function(err) {
+                setHidden(authenticatingEl, true);
+
+                setErrorState(!authenticated ? 'Failed to Authenticate with PureCloud' :
+                    'Failed to fetch/display your profile');
+            });
+
+            // ----- Helper Functions
+
+            /**
+             * Sets the base mode of the app to error and show the provided message
+             */
+            function setErrorState(errorMsg) {
+                let failureEl = document.querySelector('.failure');
+                failureEl.textContent = errorMsg;
+                setHidden(failureEl, false);
+            }
+
+            let profileLinkListener = null;
+            /**
+             * Updates the provided User Profile element state
+             * with the provided profileData
+             */
+            function renderUserProfile(profileRootEl, profileData) {
+                let linkEl = profileRootEl.querySelector('.profile-link');
+
+                if (profileLinkListener) {
+                    linkEl.removeEventListener('click', profileLinkListener);
+                    profileLinkListener = null;
+                }
+
+                if (profileData.id) {
```

Comment:
Should put this check inside the click listener so we get some logging on click and can intercept the navigation.